### PR TITLE
MINOR: [Python][CI] Ignore parquet-encryption if parquet not enabled

### PR DESCRIPTION
**Pull request by lidavidm**
Fix the failing Java/Python JNI integration build.

### CODE REVIEW

**Comment by lidavidm** [comment]
@github-actions crossbow submit conda-python-java-integration

**Comment by lidavidm** [comment]
@github-actions crossbow submit test-conda-python-java-integration

File: python/setup.py [hunk 1]
```diff
@@ -421,7 +421,8 @@ def _bundle_cython_cpp(self, name, lib_path):
     def _failure_permitted(self, name):
         if name == '_parquet' and not self.with_parquet:
             return True
-        if name == '_parquet_encryption' and not self.with_parquet_encryption:
+        if name == '_parquet_encryption' and \
+                not (self.with_parquet and self.with_parquet_encryption):
```
Comment (cyb70289) [inline python/setup.py:425]: From the name, looks to me `with_parquet_encryption == True` implies `with_parquet == True`, no?

Comment (lidavidm) [reply python/setup.py:425]: We could go that way, yes, and have one implicitly enable the other. We could also just fix the CI script not to have conflicting defaults: https://github.com/apache/arrow/blob/a32f454d86560ab357b419273bb08678e1632d30/ci/scripts/python_build.sh#L60-L61

Comment (lidavidm) [reply python/setup.py:425]: I fixed the script as well but IMO setup.py still either needs to 1) implicitly turn on Parquet if encryption is requested or 2) implicitly turn off encryption if Parquet is not enabled

**Pull request closed without merging**